mm2src/coins/utxo/rpc_clients/electrum_rpc/connection_managers/connection_context.rs

```diff
@@ -0,0 +1,60 @@
+use std::mem;
+use std::sync::{Arc, Mutex};
+
+use super::super::connection::ElectrumConnection;
+use super::super::constants::FIRST_SUSPEND_TIME;
+
+use common::now_ms;
+use keys::Address;
+
+/// A struct that encapsulates an Electrum connection and its information.
+#[derive(Debug)]
+pub struct ConnectionContext {
+    /// The electrum connection.
+    pub connection: Arc<ElectrumConnection>,
+    /// The list of addresses subscribed to the connection.
+    subs: Mutex<Vec<Address>>,
```

Comment:
You can consider using [HashSet](https://doc.rust-lang.org/std/collections/struct.HashSet.html) to avoid address duplication.